languages/java-cpg/src/main/java/de/jplag/java_cpg/transformation/matching/edges/CpgEdge.java

```diff
@@ -0,0 +1,90 @@
+package de.jplag.java_cpg.transformation.matching.edges;
+
+import static de.jplag.java_cpg.transformation.matching.edges.IEdge.EdgeCategory.*;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+import de.fraunhofer.aisec.cpg.graph.Node;
+
+/**
+ * This is a wrapper for a graph edge (with a 1:1 relation).
+ * @param <T> The type of the source node
+ * @param <R> The type of the related node
+ */
+public class CpgEdge<T extends Node, R extends Node> extends AEdge<T, R> {
```

Comment:
If you refer to IEdge vs. AEdge:
IEdge existed before AEdge, and I did not realize that it could actually just fully replace it. I'll remove the IEdge interface, then.

If you refer to the phrase "wrapper for a graph edge", then let me explain it in a bit more detail. 
To summarize: The Edge classes make up for the fact that in the CPG library, edges are represented inconsistently.
Either:
1. the edge _target node_ is a field of the source (e.g., an IfStatement node has the fields condition, thenStatement, elseStatement); the AST edge itself is not modeled
 --- or, if it is a 1:n relation ----
2. that relation is represented as a _list of nodes_ (e.g. a `FunctionType` has the field `parameters`)
3. that relation is represented as a _list of edges_ (e.g. a `Block` node has the field `statementEdges`)

The edge class of the CPG library is called `PropertyEdge`.

For the transformations, I need to _get_ and the target(s) of edges and eventually _set_ them to new values. My `Edge` classes save the appropriate getter and setter functions for all three cases. 
In the class `de.jplag.java_cpg.transformation.matching.edges.Edges`, you can see all the kinds of edges I use, and how I save the appropriate getter and setter functions from the node classes in them.